tidy3d/components/data/monitor_data.py

```diff
@@ -326,6 +327,43 @@ def normalize(self, source_spectrum_fn: Callable[[float], complex]) -> FieldData
 
         return self.copy(update=fields_norm)
 
+    def to_source(
+        self, source_time: SourceTimeType, center: Coordinate, size: Coordinate = None
+    ) -> CustomFieldSource:
+        """Create a :class:`.CustomFieldSource` from the fields stored in the :class:`.FieldData`.
+
+        Parameters
+        ----------
+        source_time: :class:`.SourceTime`
+            Specification of the source time-dependence.
+        center: Coordinate
```

Comment:
this won't render well in the docs, maybe just `Tuple[float, float, float]` works.